code/game/objects/structures/xeno.dm

```diff
@@ -695,3 +695,121 @@ TUNNEL
 	else
 		to_chat(M, "<span class='warning'>Our crawling was interrupted!</span>")
+
+//Resin Water Well
+/obj/effect/alien/resin/acidwell
+	name = "acid well"
+	desc = "An acid well. It stores acid to put out fires."
+	icon = 'icons/Xeno/acid_pool.dmi'
+	icon_state = "fullwell"
+	density = FALSE
+	opacity = FALSE
+	anchored = TRUE
+	max_integrity = 5
+	layer = RESIN_STRUCTURE_LAYER
+
+	hit_sound = "alien_resin_move"
+	destroy_sound = "alien_resin_move"
+
+	var/charges = 1
+	var/ccharging = FALSE
+	var/mob/living/carbon/xenomorph/creator = null
+
+/obj/effect/alien/resin/acidwell/Initialize()
+	. = ..()
+	update_icon()
+
+/obj/effect/alien/resin/acidwell/Destroy()
+	if(!QDELETED(creator) && creator.stat == CONSCIOUS && creator.z == z)
+		var/area/A = get_area(src)
+		if(A)
+			to_chat(creator, "<span class='xenoannounce'>You sense your acid well at [A.name] has been destroyed!</span>")
+	var/datum/effect_system/smoke_spread/xeno/acid/A = new(get_turf(src))
+	A.set_up(clamp(charges,0,2),src)
+	A.start()
+	return ..()
+
+/obj/effect/alien/resin/acidwell/examine(mob/user)
+	..()
+	if(!isxeno(user) && !isobserver(user))
+		return
+	to_chat(user, "<span class='xenoannounce'>This is an acid well made by [creator].</span>")
+
+/obj/effect/alien/resin/acidwell/deconstruct(disassembled = TRUE)
+	visible_message("<span class='danger'>[src] suddenly collapses!</span>")
+	return ..()
+
+/obj/effect/alien/resin/acidwell/update_icon()
+	..()
+	icon_state = "well[charges]"
+	set_light(charges , charges / 2, LIGHT_COLOR_GREEN)
+
+/obj/effect/alien/resin/acidwell/ex_act(severity)
+	switch(severity)
+		if(EXPLODE_DEVASTATE)
+			take_damage(210)
+		if(EXPLODE_HEAVY)
+			take_damage(140)
+		if(EXPLODE_LIGHT)
+			take_damage(70)
+
+/obj/effect/alien/resin/acidwell/attackby(obj/item/I, mob/user, params)
+	if(!isxeno(user))
+		return ..()
+	attack_alien(user)
+
+/obj/effect/alien/resin/acidwell/attack_alien(mob/living/carbon/xenomorph/M)
+	if(M.a_intent != INTENT_HARM)
+		if(charges >= 5)
+			to_chat(M, "<span class='xenoannounce'>[src] is already full!</span>")
+			return
+		if(ccharging)
+			to_chat(M, "<span class='xenoannounce'>[src] is already being filled!</span>")
+			return
+		ccharging = TRUE
+		if(!do_after(M, 10 SECONDS, FALSE, src, BUSY_ICON_BUILD))
+			ccharging = FALSE
+			return
+		if(M.plasma_stored < 200)
+			ccharging = FALSE
+			return
+		M.plasma_stored -= 200
+		charges++
+		ccharging = FALSE
+		update_icon()
+		to_chat(M,"<span class='xenonotice'>You fill up by one [src].</span>")
+	else 
+		to_chat(M, "<span class='xenowarning'>We begin removing [src]...</span>")
+		if(do_after(M, 5 SECONDS, FALSE, src, BUSY_ICON_BUILD))
+			deconstruct(FALSE)
+		return
+
+/obj/effect/alien/resin/acidwell/Crossed(atom/A)
+	. = ..()
+	if(iscarbon(A))
+		HasProximity(A)
+
+/obj/effect/alien/resin/acidwell/HasProximity(atom/movable/AM)
+	if(!iscarbon(AM))
+		return
+	if(!(AM in GLOB.alive_living_list))
+		return
```

Comment:
You gotta stop doing this `in` checks they are not good.
If your looking to check if the mob is alive, just type caste it and check `stat == DEAD`

Reply:
I believe I fixed this. Sorry about that.